gradio/processing_utils.py

```diff
@@ -410,7 +410,39 @@ def download_temp_copy_if_needed(self, url: str) -> str:
         return full_temp_file_path
 
 
-def create_tmp_copy_of_file(file_path, dir=None):
+def download_tmp_copy_of_file(
+    url_path: str, access_token: str | None = None, dir: str | None = None
+) -> tempfile._TemporaryFileWrapper:
+    if dir is not None:
+        os.makedirs(dir, exist_ok=True)
+    if access_token:
+        headers = {"Authorization": "Bearer " + access_token}
+    else:
+        headers = {}
+    file_name = Path(url_path).name
+    prefix, extension = file_name, None
+    if "." in file_name:
+        prefix = file_name[0 : file_name.index(".")]
```

Comment:
I think it would be better to use the `.stem` and `.suffix` apis from `pathlib.Path` here?

Reply:
Yes indeed, thanks @freddyaboulton!

Reply:
Cleaned up this part of the code